### PR TITLE
Adjust line property of twoslash errors to account for multi-file

### PR DESCRIPTION
**Pull request by gabro**
Fix for https://github.com/shikijs/twoslash/issues/135

### CODE REVIEW

File: packages/ts-twoslasher/test/results/multiFileErrors.json
Comment (gabro) [inline packages/ts-twoslasher/test/results/multiFileErrors.json]: this is the main change: without this PR it would (incorrectly) be `"line": 1`

File: packages/ts-twoslasher/src/index.ts
Comment (gabro) [inline packages/ts-twoslasher/src/index.ts]: this is a really naive way of doing it, I'll try to push a refactor later today

Comment (orta) [reply packages/ts-twoslasher/src/index.ts]: 👍🏻 structurally looks right though

Comment (gabro) [reply packages/ts-twoslasher/src/index.ts]: It works in most cases, but if two filenames happen to start with the same line, it will always pick the first one.

I was thinking of either:
- matching the `// filename: ` name which corresponds to `err.file!.fileName)`
- adding the line offset info directly in `nameContent` (since that's where the code is being traverse and split) so that it can be referenced here.

Any preference or alternative approaches?

**Comment by gabro** [comment]
Also @orta would you mind approving the workflow run? I'm a first-time contributor to this repo so it's currently on hold

File: packages/ts-twoslasher/src/index.ts
Comment (gabro) [reply packages/ts-twoslasher/src/index.ts]: Ok, I've pushed a refactor which implements the first proposal (so that the diff is more contained).

I've also added a test case which stresses the edge case I mentioned above (two virtual files beginning with the same line). The older implementation would yield an incorrect result, while the revised one works

Comment (orta) [reply packages/ts-twoslasher/src/index.ts]: Yeah, this looks really solid to me, nice call 👍🏻

**Comment by orta** [comment]
The fails do look legit, might be that there's another test which has a multi file error which needs a snapshot update 👍🏻

**Comment by orta** [comment]
( As you're going into the codebase anyway, can you do a server patch to the package json for twoslash, that'll get it deployed on merge )

**Comment by gabro** [comment]
> The fails do look legit, might be that there's another test which has a multi file error which needs a snapshot update 👍🏻

yep, it was in the `handlesJSON` fixture and the updated snapshot looks legit.



> ( As you're going into the codebase anyway, can you do a server patch to the package json for twoslash, that'll get it deployed on merge )

you mean bump the version? Sure, see the last commit.

**Comment by gabro** [comment]
Hi @orta can you re-run the CI? Tests should be alright now

**Review by orta — APPROVED** [review]
I can't, heh, my GitHub access is now read only also - but this all looks good to go

**Comment by gabro** [comment]
@orta oh, I just saw your tweet from Friday 😅 Congrats for all the work you've done for the TypeScript ecosystem!

Any other maintainer who can test, merge and release this?

**Comment by tonyhb** [comment]
Adding a filename to twoslash also causes this error, which the PR doesn't fix.  I'm assuming it's the same code.

EG:

```
// @filename: ./inngest/new-pr.ts
import { createFunction } from "inngest";
 
export const newPR = createFunction("New PR", "github/pull_request", ({ event }) => {
 // This function is triggered when the `github/pull_request` event is received via
 // a GH webhook.
});
```

The underlines are offset incorrectly due to filename.

**Comment by jakebailey** [comment]
Is this PR still up to date and accurate? There is a conflict, though just in the version number for the package.

**Comment by gabro** [comment]
Hi @jakebailey, the original fix still stands, I think. I haven't addressed @tonyhb's concern.
I'm not sure if I'll have time to put more work into this, but if you're interesting in emerging it I can at least resolve the conflict

**Comment by jakebailey** [comment]
I'm sure I can review it. It's just a package.json conflict, so I will quick resolve the conflicts.

**Review by jakebailey — APPROVED** [review]
LGTM but will merge/release this with the other PRs that require package publishes.

**Pull request merged**
